src/Components/Endpoints/src/DependencyInjection/HotReloadService.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection.Metadata;
+using Microsoft.Extensions.Primitives;
+
+[assembly: MetadataUpdateHandler(typeof(Microsoft.AspNetCore.Components.Endpoints.HotReloadService))]
+
+namespace Microsoft.AspNetCore.Components.Endpoints;
```

Comment:
There' already a [HotReloadManager](https://source.dot.net/#Microsoft.AspNetCore.Components/HotReload/HotReloadManager.cs,11) that's implemented using a static singleton to support hot-reload for components with an active renderer.

I opted to implement a separate service here to keep separation of concerns between endpoint-specific hot reload things.